src/main/java/org/broadinstitute/hellbender/tools/spark/transforms/markduplicates/MarkDuplicatesSparkUtils.java

```diff
@@ -80,11 +84,60 @@
                 out.add(new Tuple2<>(pair.key(header), pair));
             }
             return out;
-        }).groupByKey();
+        }).groupByKey(parallelism);
 
         return markPairedEnds(keyedPairs, scoringStrategy, finder, header);
     }
 
+    /**
+     * Like <code>groupByKey</code>, but assumes that values are already sorted by key, so no shuffle is needed,
```

Comment:
> does it guarantee that two reads with the same name but different read groups will not get grouped together?

Yes, it does. I've added a test for this as you suggested.

Do you know what read groups are used in practice? Is it OK to assume that a read pair will be placed in the same read group?